dot/state/block.go

```diff
@@ -48,6 +48,8 @@ type BlockState struct {
 	finalized     map[byte]chan<- *types.Header
 	importedLock  sync.RWMutex
 	finalizedLock sync.RWMutex
+
+	pruneKeyCh chan []common.Hash
```

Comment:
is it possible to change this to write-only? ie `chan<- []common.Hash`

also is there a reason it's a slice and not just a hash?

Reply:
I designed it to be slice since I thought keys are only pruned when they are finalized and we can process all the prune keys at one go and then sleep for some time in the Go-routine. I have removed the sleep logic and update this to be chan of hash.